A slightly different fix for "Fullscreen overlay does not resize on orientation changed"

withad opened the pull request:
Similar to [my previous fix](https://github.com/jorgesilva/TeclaAccess/pull/45) but I've now moved startFullScreenSwitchMode(500, false) (which recreates the overlay without showing the Toast) into the onConfigurationChanged method of TeclaIME, rather than having TeclaApp.onConfigurationChanged broadcast an Intent. The effect is exactly the same but the code is a little simpler.

withad committed:
Fixed bug "Fullscreen overlay does not resize on orientation changed"
(https://github.com/jorgesilva/TeclaAccess/issues/14).

Problem was caused by the fact that the fullscreen switch overlay
didn't change size/shape when the screen orientation changed. To fix this,
I've overridden onConfigurationChanged (which triggers when the orientation
changes) to send out a new intent, ACTION_RESTART_FS_SWITCH_MODE. This works
similarly to the ACTION_START_FS_SWITCH_MODE intent except that it won't
show the "Fullscreen switch enabled" toast.

To toggle the Toast, I modified mCreateSwitchRunnable to be an instance of
an inner class that takes a boolean argument, showToast. startFullScreenSwitchMode()
provides the boolean.

(Repeat of the identical commit to the master branch, as it should've gone in
develop).

withad committed:
Moving orientation changing into onConfigurationChanged in TeclaIME.

withad committed:
A slight variation on my previous bugfix for the overlay not changing size.
I hadn't noticed that there was already an onConfigurationChanged method
being used in TeclaIME. Using that to called startFullScreenSwitch(500, false)
simplifies the solution slightly, avoiding having to send an Intent from
TeclaApp.

withad committed:
Trying to debug being unable to disable Fullscreen mode.

withad committed:
Added back in the lines:

if (action.equals(TeclaApp.ACTION_STOP_FS_SWITCH_MODE)) {
	Log.d(TeclaApp.TAG, CLASS_TAG + "Received stop fullscreen switch mode intent.");
	stopFullScreenSwitchMode();
}

That I'd somehow accidentally deleted from TeclaIME's mReceiver, which was
breaking the fullscreen mode (it couldn't be turned off on its own,
only when disabling the navigation keyboard as well. Silly mistake.

ijdoc commented:
This is starting to look close to final. What I like about this solution: https://github.com/jorgesilva/TeclaAccess/pull/53 is that it does not require the boolean and custom runnable class. Could you incorporate that? I think it makes more sense to pull this branch because of the helpful documentation additions.

withad committed:
A (hopefully) final update to fix the screen orientation bug.

Everything is now contained in TeclaIME.onConfigurationChanged().
If the orientation has changed (conf.orientation != mOrientation),
mSwitchPopup is simply updated with the new height and width of the
current display.

withad commented:
That certainly seems simpler. I've updated the pull request with a slight variation of it.

I tried adding those two lines exactly but, as far as I can tell, [update()](http://developer.android.com/reference/android/widget/PopupWindow.html#update%28%29) alone won't change the current width and height, so that fix wasn't actually working. I've used [update(int width, int height)](http://developer.android.com/reference/android/widget/PopupWindow.html#update%28int,%20int%29) instead, which seems to work.

I also moved it into the existing "if (conf.orientation != mOrientation)" check, just so it doesn't needlessly trigger when the screen hasn't rotated.

ijdoc commented:
This looks good, just need to test before merging.
